snap/hooks/configure

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env bash
+
+set -eux
+
+source "${OPS_ROOT}"/helpers/snap-logger.sh "hook-configure"
+source "${OPS_ROOT}"/helpers/set-conf.sh
+
+# ----------------------------
+
+function systemd_override () {
+    DAEMON_SYSTEMD_PATH="/etc/systemd/system/snap.${PROJECT_NAME}.daemon.service.d"
+    mkdir -p "${DAEMON_SYSTEMD_PATH}"
+
+    # We are copying the service conf file patch as editing the service file generated by snapd won't survive upgrades
+    cp "${SNAP}"/ops/sys/etc/systemd/override.conf "${DAEMON_SYSTEMD_PATH}"/
+}
+
+function system_conf_override () {
+    # 1. Set the number of open file handles
+    # ulimit -n 1024 -- default in local machine
+    ulimit -n 65535
+
+    # 2. Allow the opensearch user to Disable all swap files:
+    # swapon -a -- default in local machine
+    swapoff -a || true
+
+    # 3. Ensuring sufficient virtual memory: https://www.elastic.co/guide/en/elasticsearch/reference/current/vm-max-map-count.html
+    # sysctl -w vm.max_map_count=65530 -- default in local machine
+    sysctl -w vm.max_map_count=262144
+
+    # 4. number of threads opensearch can create:
+    # should be configured automatically if opensearch ran as a service
+
+    # 5. Reduce TCP retransmission timeout = ~6 seconds
+    # sysctl -w net.ipv4.tcp_retries2=15 -- default in local machine
+    sysctl -w net.ipv4.tcp_retries2=5
+}
+
+# -------------------------------
+
+function set_access_restrictions () {
+    if [[ $# -eq 2 ]]; then
+        chmod -R "$2" "$1"
+    fi
+    chown -R snap_daemon:snap_daemon "$1"
+}
+
+function add_folder () {
+    mkdir -p "$1"
```

Comment:
using shell expression to check for the directory and later creating should be faster, something like

    [ -d "$1" ] || mkdir -p "$1"